### PR TITLE
fix grav code

### LINKED ISSUE #14784: Can't pull shit in space
### Severity

Inconvenience

### Short Description

Trying to pull shit in space is not fucking working (probably that bugfix PR)

### Reproduction

1. Preferably secure EVA gear
2. Go into space
3. Drag shit
4. Shit not being dragged???
5. Major mental breakdown
6. Major code breakdown
7. Major re-addition of a bug (or addition of a fix)

### Round ID

41812

### Test Merges

#14466 #14600 #14582

### BYOND Username

Addust

### Additional information

Inability to pull shit is pretty bad not gonna lie

### LINKED ISSUE #14786: can't drag shit like crate, canister, closet in space anymore
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: everytime i just try to drag shit in space, move up one tile, instantly stop dragging it very annoying i blame this pr #14751

 Submitted by: Bop (407757687334633473)

### LINKED ISSUE #14882: Pulling objects in no gravity with magboots bug
### Severity

Inconvenience

### Short Description

Pulling weapon recharger with active magboots in no gravity lets them fly away from you while you're pulling

### Reproduction

- no grav
- active magboots
- pull recharger
- it flies away

### Round ID

41896 approx. at 31:00 as Dylan Rathens

### BYOND Username

Aquir

### PR DESCRIPTION
**Pull request by warface1234455**
fix #14882, fix #14784, fix #14786
When you drag object in space and stop at one point, the object itself doesn't stop and still moving which is bad. Since SM can dust if hits something, this can be even worse.

ported a fix from tg
# Document the changes in your pull request
fix pulling shit in 0G being wonky

# Changelog

:cl:  

bugfix: fixed pulling shit in 0G being wonky
/:cl:

### CODE REVIEW

**Comment by ShoziOffical** [comment]
what the fuck

**Comment by ShoziOffical** [comment]
I hate turdis

**Comment by ReddicusDragon** [comment]
does this literally add a check that says if the object is movable allow it to be moved if being pulled

**Comment by Simplehorror** [comment]
didn't we do this before

**Comment by Bibby0110** [comment]
shouldn't it just be based on ProcessSpacemove of pulledby

**Comment by ShoziOffical** [comment]
it supposed to be but someone removed it

**Comment by Simplehorror** [comment]
are you sure this will work? shouldn't you just stop spaceglide in stop_pulling?

**Comment by ShoziOffical** [comment]
> are you sure this will work? shouldn't you just stop spaceglide in stop_pulling?

this will work, i have tested, no need to stop spaceglide in stop_pulling

**Comment by ShoziOffical** [comment]
@Simplehorror also thank you for adding sm bump code that instantly dusts everything it bumps to, gotta love when dragging it in 0G it dusts you if you stop at one point or will just keep moving till it dusts something

**Comment by Simplehorror** [comment]
> @Simplehorror also thank you for adding sm bump code that instantly dusts everything it bumps to, gotta love when dragging it in 0G it dusts you if you stop at one point or will just keep moving till it dusts something

ok so don't start dragging the supermatter into space?

**Comment by ShoziOffical** [comment]
> > @Simplehorror also thank you for adding sm bump code that instantly dusts everything it bumps to, gotta love when dragging it in 0G it dusts you if you stop at one point or will just keep moving till it dusts something
> 
> ok so don't start dragging the supermatter into space?

i said 0G not in space, and it is hella broken, not supposed to dust mob like that

File: code/modules/mob/living/carbon/carbon_movement.dm [hunk 1]
```diff
@@ -25,6 +25,8 @@
 	return
 	
 /mob/living/carbon/Process_Spacemove(movement_dir = 0)
+	if(..())
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_movement.dm]: replace this part with .= ..()

**Comment by ShoziOffical** [comment]
> ok so don't start dragging the supermatter into space?

you saying like you don't care about broken codes affecting gameplay

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**